### PR TITLE
#41 class specific metrics

### PR DESCRIPTION
**Pull request by Mars-Martiny**
Merge PR #96 first.

Training loop now calls `evaluate` function, the function itself has been fixed thoroughly and integrated within the pipeline (no more hallucinated arguments, cleaner saving etc.). Resolves issue #93 (finishes off issue #26)

Class specific metrics calculated in `_compute_classification_metrics` and saved for training, validation and testing steps. Resolves issue #41 

Expanded `RunSaver`, now tracks test set evaluation metrics and plots of class specific metrics along with confusion matrix for training, validation and testing steps. Concludes issue #24

### CODE REVIEW

**Review by AndreHrs — APPROVED** [review]
Approved, I'm not sure if this fixes issue #83 by any chance but I will have a look later after integrating the branch
